modules/solid_mechanics/include/materials/EigenDecompMaterial.h

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+#include "RankTwoTensorForward.h"
+/**
+ * Perform eigen decomposition on Rank two tensor
+ */
+template <bool is_ad>
+class EigenDecompMaterialTempl : public Material
+{
+public:
+  static InputParameters validParams();
+
+  EigenDecompMaterialTempl(const InputParameters & parameters);
+
+  virtual void computeQpProperties() override;
+
+protected:
```

Comment:
```suggestion
protected:
    virtual void computeQpProperties() override;
```

this method is protected in the base class and should stay that way